### PR TITLE
use Github Actions for CI

### PR DESCRIPTION
**Pull request by bvreede**
Travis does not work ~at the moment~ anymore; this PR would incorporate a GA workflow that runs the tests on pull requests.

I am not sure about the codecov functionality in this workflow; there is a report produced here in the PR (this only happened when I changed a test) but it does not show actual coverage. Not sure why though... Help appreciated.

closes https://github.com/eWaterCycle/era5cli/issues/65

### CODE REVIEW

File: .github/workflows/test_codecov.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+on:
+  pull_request
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+    name: Run tests for ${{ matrix.python-version }}
```
Comment (Peter9192) [inline .github/workflows/test_codecov.yml:12]: Why did you remove the fail fast option again?

File: .github/workflows/test_codecov.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        architecture: x64
+    - name: Install dependencies and package
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+        pip install -r requirements-dev.txt
+        pip install -e .
```
Comment (Peter9192) [inline .github/workflows/test_codecov.yml:25]: Do we really need 4 lines to install?

File: .github/workflows/test_codecov.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    - name: Test with pytest
+      run: pytest
+    - name: Flake8
+      run: flake8
+    - name: Test coverage
+      run: |
+        py.test --cov=era5cli --cov-report term --cov-report xml:cov.sml tests/
```
Comment (Peter9192) [inline .github/workflows/test_codecov.yml:32]: Doesn't this work if you #dropthedot? https://stackoverflow.com/a/41893170

File: .github/workflows/test_codecov.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1.3.2
+      with:
+        files: cov.sml
+        directory: ./coverage/reports/
+        env_vars: python-version
```
Comment (Peter9192) [inline .github/workflows/test_codecov.yml:38]: Do you expect a difference for different python versions? Or is it just easier to keep it in the same loop?

File: .travis.yml [hunk 5]
```diff
@@ -1,14 +0,0 @@
-language: python
```
Comment (Peter9192) [inline .travis.yml:1]: I wonder why the travis checks are still showing on this PR.

**Review by Peter9192 — COMMENTED** [review]
Hey @bvreede that's a really nice PR you made here. Honestly I haven't that much experience with setting up actions, so I just entered "just ask everything mode". If you can answer my questions, particularly about travis still showing, I'm happy to approve it.

File: .github/workflows/test_codecov.yml (hunk 1 above)
Comment (bvreede) [reply .github/workflows/test_codecov.yml:12]: It crashed the workflow.. but that was a few commits ago. I can check it again (or, eh, google it :D)

**Comment by nielsdrost** [comment]
@bvreede I've updated the branch protection rule for the master branch not to require the Travis checks anymore. :-)

File: .travis.yml (hunk 5 above)
Comment (bvreede) [reply .travis.yml:1]: I think because they're still in master?

**Review by Peter9192 — APPROVED** [review]
Now that Niels fixed the branch protection I'm happy to see it merged :-).

Of course I still welcome your answers to my other questions.

File: .github/workflows/test_codecov.yml (hunk 4 above)
Comment (bvreede) [reply .github/workflows/test_codecov.yml:38]: I don't expect differences, but it's useful to keep the info there I expect.

File: .github/workflows/test_codecov.yml (hunk 3 above)
Comment (bvreede) [reply .github/workflows/test_codecov.yml:32]: This was copied verbatim from the Travis file. Dropping it!

File: .github/workflows/test_codecov.yml (hunk 2 above)
Comment (bvreede) [reply .github/workflows/test_codecov.yml:25]: ... Maybe? One for upgrading, one for dependencies, one for development tools (eg pytest, flake8), one for the package itself — can we wrap this into one?
I am not that fluent in pip commands (this is the result of copy pasting) and this works so i'm happy to leave it, but also very open to suggestions.

**Comment by bvreede** [comment]
Thanks @nielsdrost and @Peter9192 — merging!
